vault/request_handling.go

```diff
@@ -875,6 +879,12 @@ func (c *Core) isLoginRequest(ctx context.Context, req *logical.Request) bool {
 func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp *logical.Response, retAuth *logical.Auth, retErr error) {
 	defer metrics.MeasureSince([]string{"core", "handle_request"}, time.Now())
 
+	// Check for request role
+	var role string
+	if reqRole := ctx.Value(logical.CtxKeyRequestRole{}); reqRole != nil {
+		role = reqRole.(string)
+	}
+
```

Comment:
Can we move this down just above `RegisterAuth`? That seems to be the only place it's used in this method.

Reply:
good call, I also moved it here https://github.com/hashicorp/vault/pull/22583/files#diff-d9a68017e2e6bafd870acc36e97049b62487de046f4cc8541921ef66b30d49b5R794